services/intersection-api/api/src/main/java/us/dot/its/jpo/ode/api/services/SNMPService.java

```diff
@@ -0,0 +1,169 @@
+import org.snmp4j.security.USM;
+import org.snmp4j.security.UsmUser;
+import org.snmp4j.smi.*;
+import org.snmp4j.transport.DefaultUdpTransportMapping;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+
+import javax.annotation.PostConstruct;
+
+@Service
+public class SNMPService {
+
+    private static final Logger logger = LoggerFactory.getLogger(SNMPService.class);
+
+    @PostConstruct
+    public void init() throws Exception {
+
+    }
+
+    // Some RSUs do not require authentication to retrieve information. They may
+    // respond to SNMP V2 requests.
+    public String getSnmpV2Value(String ipAddress, String community, String oid) throws Exception {
+
+        TransportMapping<UdpAddress> transport = new DefaultUdpTransportMapping();
+        Snmp snmp = new Snmp(transport);
+        transport.listen();
+
+        CommunityTarget<UdpAddress> target = new CommunityTarget<>();
+        target.setCommunity(new OctetString(community));
+        target.setAddress(new UdpAddress(ipAddress + "/161"));
+        target.setRetries(2);
+        target.setTimeout(1500);
+        target.setVersion(SnmpConstants.version2c);
+
+        PDU pdu = new PDU();
+        pdu.add(new VariableBinding(new OID(oid)));
+        pdu.setType(PDU.GET);
+
+        ResponseEvent<UdpAddress> responseEvent = snmp.send(pdu, target);
+        snmp.close();
+
+        if (responseEvent != null && responseEvent.getResponse() != null) {
+            VariableBinding vb = responseEvent.getResponse().get(0);
+            return vb.getVariable().toString();
+        } else {
+            throw new RuntimeException("SNMP GET timed out or returned null.");
+        }
+
+    }
+
+    public Variable getSnmpV3Value(String ip, String username, String authPass, String privPass, String oid)
+            throws IOException {
+
+        TransportMapping<UdpAddress> transport = new DefaultUdpTransportMapping();
+        Snmp snmp = new Snmp(transport);
+        transport.listen();
+
+        // Add USM and user
+        USM usm = new USM(
+                SecurityProtocols.getInstance(),
+                new OctetString(MPv3.createLocalEngineID()),
+                0);
+        SecurityModels.getInstance().addSecurityModel(usm);
+
+        // Add Required Security Protocols
+        SecurityProtocols.getInstance().addAuthenticationProtocol(new AuthSHA());
+        SecurityProtocols.getInstance().addPrivacyProtocol(new PrivAES128());
+
+        snmp.getUSM().addUser(
+                new OctetString(username),
+                new UsmUser(
+                        new OctetString(username),
+                        AuthSHA.ID, new OctetString(authPass),
+                        PrivAES128.ID, new OctetString(privPass)));
+
+        // Configure the target
+        UserTarget<UdpAddress> target = new UserTarget<>();
+        target.setAddress(new UdpAddress(ip + "/161"));
+        target.setRetries(2);
+        target.setTimeout(3000);
```

Comment:
(blocking) these configurations should be set and configured in the `application.yaml` in case we need to adjust them.
Also, is there a reason the timeout is different from the V2 query?